src/common/java/net/minecraftforge/gradle/common/task/DownloadAssets.java

```diff
@@ -48,21 +49,32 @@ public void run() throws IOException, InterruptedException {
         AssetIndex index = Utils.loadJson(getIndex(), AssetIndex.class);
         List<String> keys = new ArrayList<>(index.objects.keySet());
         Collections.sort(keys);
-        ExecutorService executorService = Executors.newFixedThreadPool(16);
-        CopyOnWriteArrayList<String> downloadingFailedURL = new CopyOnWriteArrayList<>();
+        File assetsPath = new File(Utils.getMCDir(), "/assets/objects");
+        ExecutorService executorService = Executors.newFixedThreadPool(8);
```

Comment:
I recently read Loom's Code, Maybe We can increase download Threads depends on the user's processer.
like:
```
ExecutorService executorService = Executors.newFixedThreadPool(Math.min(8, Math.max(Runtime.getRuntime().availableProcessors() / 2, 1)));
```
https://github.com/FabricMC/fabric-loom/blob/8b1268abe741d9c578a654c6a82b9b7788c43bb9/src/main/java/net/fabricmc/loom/providers/MinecraftAssetsProvider.java#L87